metrics.go

```diff
@@ -63,6 +96,21 @@ func MeasureSince(key []string, start time.Time) {
 //     SetGauge([]string{"my","namespace"}, 2.0)
 func SetGauge(key []string, val float32) {
 	if metriks != nil {
+		normalizeKeys(key)
 		metriks.SetGauge(key, val)
 	}
 }
+
+// This function is used to make metric names safe for all metric services. Specifically, prometheus does
+// not support * or / in metric names.
+func normalizeKeys(key []string) {
+	if key != nil {
+		for i, k := range(key) {
+			// first replace */* with all
+			k = strings.Replace(k, allMatcher, allReplacement, -1)
+
+			// now replace all other invalid characters with a safe one.
+			key[i] = invalidCharactersRE.ReplaceAllString(k, normalizedToken)
```

Comment:
See comment on the regexp, this should be a test to see whether the character is in a set of white listed characters.